File: fixtures/scale/dart/large/lib/ciphers/gronsfeld/gronsfeld_key.dart
class GronsfeldKey {
  final String digits;

  GronsfeldKey(this.digits);

  int digitAt(int position) => digits.codeUnitAt(position % digits.length) - '0'.codeUnitAt(0);

  factory GronsfeldKey.defaultKey() => GronsfeldKey('31415');
}

File: fixtures/scale/kotlin/large/src/cryptobench/ciphers/gronsfeld/GronsfeldKey.kt
package cryptobench.ciphers.gronsfeld

data class GronsfeldKey(
    val digits: String,
) {
    fun digitAt(position: Int): Int = digits[position % digits.length] - '0'

    companion object {
        fun default(): GronsfeldKey = GronsfeldKey("31415")
    }
}
